inf-clojure.el

```diff
@@ -676,14 +676,22 @@ to continue it."
   "Remove subprompts from STRING."
   (replace-regexp-in-string inf-clojure-subprompt "" string))
 
+(defvar inf-clojure--previous-response-ended-with-prompt nil)
+
 (defun inf-clojure-preoutput-filter (str)
   "Preprocess the output STR from interactive commands."
   (inf-clojure--log-string str "<-RES----")
-  (cond
-   ((string-prefix-p "inf-clojure-" (symbol-name (or this-command last-command)))
-    ;; Remove subprompts and prepend a newline to the output string
-    (inf-clojure-chomp (concat "\n" (inf-clojure-remove-subprompts str))))
-   (t str)))
+  (let* ((last-line (car (last (split-string str "\n"))))
+         (response-was-one-line (string= last-line str))
```

Comment:
I would appreciate some elisp feedback here.  I could get the same result by checking the length of the split str but I'm not sure if there's a scenario where this is a more valid thing to do.  I mostly just did it because I didn't like the idea of another let binding for the split str.

Reply:
Probably it makes a bit more sense first to check if something was multiline and only then split it.